crates/bevy_reflect/src/impls/bevy_ecs.rs

```diff
@@ -38,6 +39,17 @@ impl ReflectComponent {
         (self.reflect_component)(archetype, entity_index)
     }
 
+    /// # Safety
+    /// This does not do bound checks on entity_index. You must make sure entity_index is within bounds before calling.
+    /// To avoid having multiple mutable pointers to the same data, this method can only be called in a thread-local system.
```

Comment:
Its good to call out the "scheduler" side of things in the "Safety", but we should also call out generically that this could violate rust's mutability rules. Callers should verify that they are not violating the rules.